frontend/src/components/CategoryWrapper.vue

```diff
@@ -9,10 +9,15 @@ const props = defineProps<{
   shouldShowButton: boolean
 }>()
 
+const emit = defineEmits<{
+  toggled: [value: boolean]
+}>()
+
```

Comment:
Hm meiner Meinung nach ist dieser emit hier fehl am platz. Die eizige Stelle an der auf den emit gehört von außern ist TextEditorCategory, um den TextEditor zu fokussieren sobald er sichtbar wird. Es wäre meiner Meinung nach sauberer, einfach in der onMounted Methode von TextEditor den das htmlelement zu fokussieren und den ganzen code zum emit hier rauszunehmen, das brauchen wir nicht (bzw wird sogar nur dafür genutzt, was eine einfache onMounted methode abdecken könnte ohne durch 3 komponenten zu gehen).

Ich würde an dieser Stelle einfach nur sowas hier machen:

```
const props = defineProps<{
  classes?: string
  label: string
  shouldShowButton: boolean
}>()

const shouldShowButton = ref<boolean>(props.shouldShowButton)
</script>

<template>
  <TextButton
    v-if="shouldShowButton"
    button-type="tertiary"
    :class="classes"
    data-testId="category-wrapper-button"
    :icon="IconAdd"
    :label="label"
    size="small"
    @click="shouldShowButton = !shouldShowButton"
  />
  <div v-else data-testId="category-wrapper-component"><slot /></div>
</template>
```